### ISSUE TITLE
hbsfy with --traverse fails with inline partials

### ISSUE THREAD

**Issue by kentmw** [epeli/node-hbsfy#54]
With Handlebars 4.0.0, they introduced inline partials: http://handlebarsjs.com/partials.html#inline-partials

Our project added the --traverse flag so that we could bundle our partials in our browserify bundles (very helpful!).

These don't seem to mix as the inline partials are defined inline inside the same template.

Take this example from the handlebars documentation:
```
{{#*inline "myPartial"}}
  My Content
{{/inline}}
{{#each children}}
  {{> myPartial}}
{{/each}}
```
In this example myPartial will assumed to be node_module dependency. My thoughts is it would be nice to be able to give an indication to the hbsfy parser that this is an inline partial and not modify it.

Something like:

```
{{#*inline "myPartial"}}
  My Content
{{/inline}}
{{#each children}}
  {{> inline:myPartial}}
{{/each}}
```
I believe this would just require:
```
if (p.indexOf('inline:') !== 0) {
   ...
}
```
on line https://github.com/epeli/node-hbsfy/blob/master/index.js#L139

I'll make a pull request if we think this the way to solve this.

**Comment by mandragorn** [comment]
:+1:

**Comment by kentmw** [comment]
Jk, made the pull request anyway. If anyone wants to check it out, I'd appreciate it.

**Comment by kentmw** [comment]
I'm working out the fact that you still have to revert the name of the partial back to its non-prefixed state after skipping the commonjs aspect.

**Comment by kentmw** [comment]
Well, without attempt so just in time updates to the `file` or `js` that comes from your templates on this line: https://github.com/epeli/node-hbsfy/blob/master/index.js#L132, it seems like the way to do it is just have the same prefix for the inline partial.

```
{{#*inline "inline:myPartial"}}
  My Content
{{/inline}}
{{#each children}}
  {{> inline:myPartial}}
{{/each}}
```

**Comment by kentmw** [comment]
This is the pull request: https://github.com/epeli/node-hbsfy/pull/55

**Comment by kirbysayshi** [comment]
@kentmw thank you so much for the initiative with the PR! Also for reporting this, I had no idea handlebars added this.

While your `inline:` suggestion would definitely be future proof, I'm concerned about introducing new syntax to handlebars that only exists in this plugin.

What do you think about instead searching the template for `*inline` statements when a `PartialStatement` is found, and if an `*inline` is indeed found, abort the hbsfy traverse logic? That would make the `inline` logic more of an override, while not introducing new syntax and preserving what users would like expect.

**Comment by kentmw** [comment]
@kirbysayshi updated my PR to be transparent with no new syntax. Also added tests.

**Comment by kentmw** [comment]
@kirbysayshi I think it's ready for your review.

**Comment by kentmw** [comment]
@kirbysayshi  sorry to be persistent, but I was hoping you could take a look at the PR this week. Our project is blocked until this is resolved. Hopefully it just needs a merge and re-release

**Comment by kirbysayshi** [comment]
If you're using npm, you can always refer to your github fork in your package.json so to not be blocked by PRs to public packages

**Comment by kentmw** [comment]
@kirbysayshi thanks for the tip. Pretty sure the company I'm working with wouldn't be too happy moving their project to depend on my github fork, but I'll give it a go. I appreciate the thought.

**Comment by kirbysayshi** [comment]
Should be fixed in v2.7.0, published.

**Comment by kentmw** [comment]
Thank you!!
